python/xorbits/deploy/kubernetes/tests/test_kubernetes.py

```diff
@@ -221,3 +225,59 @@ def gen_data(n=100):
 
         res = xr.spawn(gen_data).to_object()
         print(res)
+
+
+@pytest.mark.skipif(not kube_available, reason="Cannot run without kubernetes")
+@pytest.mark.asyncio
+async def test_request_workers():
+    with _start_kube_cluster(
+        supervisor_cpu=0.2,
+        supervisor_mem="1G",
+        worker_cpu=0.2,
+        worker_mem="1G",
+        worker_cache_mem="64m",
+        use_local_image=True,
+    ) as cluster_client:
+        cluster_api = WebClusterAPI(address=cluster_client.endpoint)
+        with pytest.raises(ValueError) as exc:
+            await cluster_api.request_workers(worker_num=1, timeout=-10)
+        assert exc.type == ValueError
+        assert "Please specify a `timeout` that is greater than zero" in str(exc.value)
+        with pytest.raises(ValueError) as exc:
+            await cluster_api.request_workers(worker_num=-10, timeout=1)
+        assert exc.type == ValueError
+        assert "Please specify a `worker_num` that is greater than zero" in str(
+            exc.value
+        )
+        with pytest.raises(ValueError) as exc:
+            await cluster_api.request_workers(worker_num=0, timeout=1)
+        assert exc.type == ValueError
+        assert "Please specify a `worker_num` that is greater than zero" in str(
+            exc.value
+        )
+        new_workers = await cluster_api.request_workers(worker_num=1, timeout=300)
+        assert len(new_workers) == 1
+        with pytest.raises(TimeoutError) as exc:
+            await cluster_api.request_workers(worker_num=1, timeout=0)
+        assert exc.type == TimeoutError
+        assert "Request worker timeout" in str(exc.value)
+        simple_job()
+
+
+@pytest.mark.skipif(not kube_available, reason="Cannot run without kubernetes")
+@pytest.mark.asyncio
+async def test_request_workers_insufficient():
+    with _start_kube_cluster(
+        supervisor_cpu=0.5,
+        supervisor_mem="1G",
+        worker_cpu=0.5,
+        worker_mem="1G",
+        worker_cache_mem="64m",
+        use_local_image=True,
+    ) as cluster_client:
+        cluster_api = WebClusterAPI(address=cluster_client.endpoint)
+        with pytest.raises(MemoryError) as exc:
+            await cluster_api.request_workers(worker_num=1, timeout=30)
+        assert exc.type == MemoryError
+        assert "Insufficient cpu" in str(exc.value)
+        simple_job()
```

Comment:
Since you cannot request workers sucessfully, you can skip the simple job test here.